sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/EstimationUtils.scala

```diff
@@ -128,6 +149,25 @@ object EstimationUtils {
     if (outputRowCount > 0) outputRowCount * getSizePerRow(attributes, attrStats) else 1
   }
 
+  /**
+   * Get the size in bytes of a value of a constant expression
+   * @param expr
+   * @return the size in bytes
+   */
+  def getConstantLen(expr: Expression): Long = {
```

Comment:
Use "value"? Otherwise the expression needs to be evaluated twice.

Reply:
done